R/getADMBHessian.R

```diff
@@ -3,32 +3,40 @@
 #' This function reads in all of the information contained in the
 #' .hes file. Some is needed for relaxing the covariance matrix, while the rest
 #' is recorded and rewritten to file as ADMB expects.
-#' @param dir Directory in which .hes file is located. Defaults to the working
-#'  directory.
-#' @param File Deprecated. Use `dir` instead.
-#' @param FileName Name of .hes file. Defaults to admodel.hes.
+#' @param hesfile Name of .hes file, including the full path (can be
+#' relative to working directory).
+#' @param File Deprecated. Add path to `hesfile` input instead.
+#' @param FileName Deprecated. Use `hesfile`` instead.
 #' @return A list with elements num.pars, hes, hybrid_bounded_flag, and scale.
 #' @author Cole Monnahan
 #' @export
 #' @seealso [read.admbFit()], [NegLogInt_Fn()]
 #' @note Explanation of the methods (in PDF form):
 #' <https://github.com/admb-project/admb-examples/blob/master/admb-tricks/covariance-calculations/ADMB_Covariance_Calculations.pdf>
-getADMBHessian <- function(dir = getwd(),
+getADMBHessian <- function(hesfile = "admodel.hes",
                            File = lifecycle::deprecated(),
-                           FileName = "admodel.hes") {
+                           FileName = lifecycle::deprecated()) {
 
   # deprecated variable warnings -----
   # soft deprecated for now, but fully deprecate in the future.
   if (lifecycle::is_present(File)) {
     lifecycle::deprecate_warn(
       when = "1.46.0",
       what = "getADMBHessian(File)",
-      details = "Please use 'dir' instead"
+      details = "Please use 'hesfile' instead"
     )
-    dir <- File
+    File = getwd()
```

Comment:
Hmm, I think this line `File = getwd()` should be removed, as should line 36. I'm not sure how would be best to construct the output of hesfile... maybe only construct it if both File and FileName are present (using another `if_present() statement)?

Reply:
Good idea. Feel free to check if the change matches what you were thinking.